File: 8vim/src/main/java/inc/flide/vim8/keyboardActionListners/ButtonKeypadActionListener.java
package inc.flide.vim8.keyboardActionListners;

import android.view.View;

import com.hijamoya.keyboardview.KeyboardView;

import inc.flide.vim8.MainInputMethodService;

public class ButtonKeypadActionListener extends KeypadActionListener implements KeyboardView.OnKeyboardActionListener {
    public ButtonKeypadActionListener(MainInputMethodService mainInputMethodService, View view) {
        super(mainInputMethodService, view);
    }

    @Override
    public void onKey(int primaryCode, int[] keyCodes) {
        super.handleInputKey(primaryCode, 0);
    }

    @Override
    public void swipeLeft() {
    }

    @Override
    public void swipeRight() {
    }

    @Override
    public void swipeDown() {
        mainInputMethodService.hideKeyboard();
    }

    @Override
    public void swipeUp() {
    }

    @Override
    public void onPress(int primaryCode) {
    }

    @Override
    public void onRelease(int primaryCode) {
    }
}

File: 8vim/src/main/java/inc/flide/vim8/keyboardActionListners/ButtonKeypadActionListener.kt
package inc.flide.vim8.keyboardActionListners

import android.inputmethodservice.KeyboardView.OnKeyboardActionListener
import android.view.View
import inc.flide.vim8.MainInputMethodService

class ButtonKeypadActionListener(view: View) : KeypadActionListener(view), OnKeyboardActionListener {
    override fun onKey(primaryCode: Int, keyCodes: IntArray?) {
        super.handleInputKey(primaryCode, 0)
    }

    override fun onText(text: CharSequence?) {
        super.onText(text)
    }
    override fun swipeLeft() {}
    override fun swipeRight() {}
    override fun swipeDown() {
        MainInputMethodService.hideKeyboard()
    }

    override fun swipeUp() {}
    override fun onPress(primaryCode: Int) {}
    override fun onRelease(primaryCode: Int) {}
}